var/spack/repos/builtin/packages/prng/package.py

```diff
@@ -0,0 +1,42 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Prng(AutotoolsPackage):
+    """Pseudo-Random Number Generator library."""
+
+    homepage = "http://statmath.wu.ac.at/prng/"
+    url      = "http://statmath.wu.ac.at/prng/prng-3.0.2.tar.gz"
+    list_url = "http://statmath.wu.ac.at/prng"
+
+    version('3.0.2', '80cb0870f2d18618bd2772f9e1dc1a70')
+
+    patch('prng-3.0.2-shared.patch', when="@3.0.2")
+    patch('prng-3.0.2-fix-c99-inline-semantics.patch', when="@3.0.2")
```

Comment:
Have these patches been submitted upstream? Or did they come from upstream?

Reply:
They've been submitted with this PR for the first time, why?

Reply:
A lot of developers get upset when people discover bugs in their software and fix those bugs but don't tell them. I just want to make sure the developers of `prng` are aware of these bugs and have these patches. It's also a good way to know that the developers looked at the patches and ensure that they are safe and valid and don't create new bugs.

Reply:
Ok I get it. I took the Gentoo recipe as a reference (including the patches) so I cannot ensure whether the developers are already aware. I will double check in any case.

Thanks for the explanation.